common/workunit/workunit.cpp

```diff
@@ -1712,6 +1712,207 @@ class GraphScopeIterator : public CInterfaceOf<IConstWUScopeIterator>
+            scratchpad.append("]");
+            return scratchpad.str();
+        }
+        case WaIsScheduled: return trueToStr(wf->isScheduled());
+        case WaIdSuccess: return queryWfid(wf->querySuccess(), scratchpad);
+        case WaIdFailure: return queryWfid(wf->queryFailure(), scratchpad);
+        case WaIdRecovery: return queryWfid(wf->queryRecovery(), scratchpad);
+        case WaIdPersist: return queryWfid(wf->queryPersistWfid(), scratchpad);
+        case WaIdScheduled: return queryWfid(wf->queryScheduledWfid(), scratchpad);
+        case WaPersistName: return queryOptString(wf->getPersistName(adaptor));
+        case WaLabel: return queryOptString(wf->getLabel(adaptor));
+        case WaMode: return wf->queryMode() != WFModeNormal ? queryWorkflowModeText(wf->queryMode()) : nullptr;
+        case WaType: return wf->queryType() != WFTypeNormal ? queryWorkflowTypeText(wf->queryType()) : nullptr;
+        case WaState: return queryWorkflowStateText(wf->queryState());
+        case WaCluster: return queryOptString(wf->queryCluster(adaptor));
+        case WaCriticalSection: return queryOptString(wf->getCriticalName(adaptor));
+        /*
+        The followng attributes are not generated - I'm not convinced they are very useful, but they could be added later.
+        virtual bool isScheduledNow() const = 0;
+        virtual IWorkflowEvent * getScheduleEvent() const = 0;
+        virtual unsigned querySchedulePriority() const = 0;
+        virtual bool hasScheduleCount() const = 0;
+        virtual unsigned queryScheduleCount() const = 0;
+        virtual unsigned queryRetriesAllowed() const = 0;
+        virtual int queryPersistCopies() const = 0;  // 0 - unmangled name,  < 0 - use default, > 0 - max number
+        virtual bool queryPersistRefresh() const = 0;
+        virtual unsigned queryScheduleCountRemaining() const = 0;
+        virtual unsigned queryRetriesRemaining() const = 0;
+        virtual int queryFailCode() const = 0;
+        virtual const char * queryFailMessage() const = 0;
+        virtual const char * queryEventName() const = 0;
+        virtual const char * queryEventExtra() const = 0;
+        */
+        }
+        return nullptr;
+    }
+
+    virtual const char * queryHint(const char * kind) const
+    {
+        return nullptr;
+    }
+
+protected:
+    bool initWorkflowItem()
+    {
+        curScope.clear().append(WorkflowScopePrefix).append(workflow.item(curWorkflow).queryWfid());
+        return true;
+    }
+
+    const char * queryValue(unsigned value, StringBuffer & scratchpad, const char * prefix = nullptr) const
+    {
+        return scratchpad.clear().append(prefix).append(value).str();
+    }
+
+    const char * queryOptString(IStringVal & value) const
+    {
+        const char * text = value.str();
+        if (!text || !*text)
+            return nullptr;
+        return text;
+    }
+
+    const char * queryWfid(unsigned value, StringBuffer & scratchpad) const
```

Comment:
Worth renaming this function too?  It seems to be generating the wfid string rather than querying the value of member variable etc.